### ISSUE TITLE
Freeze at "In Progress"

### ISSUE THREAD

**Issue by kycfree1** [flozz/yoga-image-optimizer#30]
Conversion status freezes at "In Progress". Conversions are never completed.

The issue only occurred since the latest 1.1.2 update (flatpak)

Encountered the same issue on fully updated versions of both PopOs & Manjaro.

**Comment by flozz** [comment]
Hello,

Thank you for the report! Can you run the app from a terminal to see if there is some logs?

To run the Flatpak version from the cli, the command is:

    flatpak run org.flozz.yoga-image-optimizer

**Comment by flozz** [comment]
Thank you for the logs.

It seems that an error occurred during the optimization of one of the images, and when the thread returns, the main process crash because the output file is missing.

The errors raised in optimization threads were not handled at all by YOGA Image Optimizer 1.1.2, so it is not possible to know what is the problem (but probably a corrupted image or maybe the system had gone out of memory). I already improved this on the `master` branch: now errors in optimization thread are reported in the logs and it does not crash the whole optimization process anymore.

TL;DR: The next version of YOGA Image Optimizer (and the current `master` branch) should fix your problem. I have some more fix to work on but I hope being able to release a new version of YOGA Image Optimizer soon :)

**Comment by kianmeng** [comment]
Maybe this is a similar issue? Error encountered in Python 3.7.14. No issue in 3.10.7 though. Manage different Python versions using `pyenv`.

Steps to reproduce:

```sh
git clone https://github.com/flozz/yoga-image-optimizer.git
cd yoga-image-optimizer
sudo pip3 install .
```

Stacktrace:

```sh
$ yoga-image-optimizer
...
Exception in thread QueueManagerThread:
Traceback (most recent call last):
  File "/home/foobar/.pyenv/versions/3.7.14/lib/python3.7/threading.py", line 926, in _bootstrap_inner                                                                                     self.run()
  File "/home/foobar/.pyenv/versions/3.7.14/lib/python3.7/threading.py", line 870, in run self._target(*self._args, **self._kwargs)                                               File "/home/foobar/.pyenv/versions/3.7.14/lib/python3.7/concurrent/futures/process.py", line 375, in _queue_management_worker
    thread_wakeup.clear()                                                                   File "/home/foobar/.pyenv/versions/3.7.14/lib/python3.7/concurrent/futures/process.py", line 93, in clear
    while self._reader.poll():
  File "/home/foobar/.pyenv/versions/3.7.14/lib/python3.7/multiprocessing/connection.py", line 255, in poll
    self._check_closed()
  File "/home/foobar/.pyenv/versions/3.7.14/lib/python3.7/multiprocessing/connection.py", line 136, in _check_closed
    raise OSError("handle is closed")
OSError: handle is closed

Traceback (most recent call last):
  File "/home/foobar/.pyenv/versions/3.7.14/lib/python3.7/site-packages/yoga_image_optimizer/main_window.py", line 385, in _on_main_window_destroyed
    app.quit()
  File "/home/foobar/.pyenv/versions/3.7.14/lib/python3.7/site-packages/yoga_image_optimizer/application.py", line 142, in quit
    self._thumbnailer.cancel_all()
  File "/home/foobar/.pyenv/versions/3.7.14/lib/python3.7/site-packages/yoga_image_optimizer/thumbnailer.py", line 156, in cancel_all
    self._executor.shutdown(wait=False, cancel_futures=True)
TypeError: shutdown() got an unexpected keyword argument 'cancel_futures'
```

**Comment by flozz** [comment]
@kianmeng It is a different error, probably related to API changes in `concurrent.futures`. I opened a new issue to track this bug. Nice catch! :)

**Comment by flozz** [comment]
This error

```
/usr/include/c++/12.1.0/bits/stl_vector.h:1123: std::vector<_Tp, _Alloc>::reference std::vector<_Tp, _Alloc>::operator [with _Tp = float; _Alloc = std::allocator; reference = float&; size_type = long unsigned int]: Assertion '__n < this->size()' failed.
```

Was in fact caused by a bug in the Gutezli library and the assertion error occurred only in the Flatpak package. More information in this other bug:

https://github.com/flozz/yoga-image-optimizer/issues/35

The Guetzli bug is now fixed, Pygutzli was patched and updated and the YOGA Image Optimizer Flatpak package has been updated :)
